src/System.DirectoryServices.Protocols/tests/DirectoryAttributeTests.cs

```diff
@@ -0,0 +1,349 @@
+            string[] values = new string[] { "value1", null, "value2" };
+            Assert.Throws<ArgumentNullException>("value", () => new DirectoryAttribute("Name", values));
+        }
+
+        [Fact]
+        public void Ctor_InvalidObjectInValues_ThrowsArgumentException()
+        {
+            object[] values = new object[] { 1 };
+            Assert.Throws<ArgumentException>("value", () => new DirectoryAttribute("Name", values));
+        }
+
+        [Fact]
+        public void Name_Set_GetReturnsExpected()
+        {
+            var attribute = new DirectoryAttribute { Name = "Name" };
+            Assert.Equal("Name", attribute.Name);
+        }
+
+        [Fact]
+        public void Name_SetNull_ThrowsArgumentNullException()
+        {
+            var attribute = new DirectoryAttribute();
+            Assert.Throws<ArgumentNullException>("value", () => attribute.Name = null);
+        }
+
+        [Fact]
+        public void GetValues_Mixed_Success()
+        {
+            var attribute = new DirectoryAttribute { "abc", new byte[] { 100, 101, 102 } };
+            Assert.Equal(new byte[][] { new byte[] { 97, 98, 99 }, new byte[] { 100, 101, 102 } }, attribute.GetValues(typeof(byte[])));
+            Assert.Equal(new string[] { "abc", "def" }, attribute.GetValues(typeof(string)));
+        }
+
+        [Fact]
+        public void GetValues_ContainsUri_ThrowsNotSupportedException()
+        {
+            var attribute = new DirectoryAttribute { "abc", new byte[] { 100, 101, 102 }, new Uri("http://microsoft.com") };
+            Assert.Throws<NotSupportedException>(() => attribute.GetValues(typeof(byte[])));
+            Assert.Throws<NotSupportedException>(() => attribute.GetValues(typeof(string)));
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(typeof(Uri))]
+        [InlineData(typeof(int))]
+        public void GetValues_InvalidType_ThrowsArgumentException(Type valuesType)
+        {
+            var attribute = new DirectoryAttribute();
+            Assert.Throws<ArgumentException>("valuesType", () => attribute.GetValues(valuesType));
+        }
+
+        [Fact]
+        public void Indexer_SetMultipleType_GetReturnsExpected()
+        {
+            var attribute = new DirectoryAttribute { "value" };
+
+            attribute[0] = new Uri("http://microsoft.com");
+            Assert.Equal(new Uri("http://microsoft.com"), attribute[0]);
+
+            attribute[0] = "value";
+            Assert.Equal("value", attribute[0]);
+
+            attribute[0] = new byte[] { 1, 2, 3 };
+            Assert.Equal(new byte[] { 1, 2, 3 }, attribute[0]);
+        }
+
+        [Fact]
+        public void Indexer_SearchResult_GetReturnsExpected()
+        {
+            var attribute = new DirectoryAttribute { "value", new byte[] { 97, 98, 99 }, new byte[] { 240, 144, 128 } };
+            FieldInfo field = attribute.GetType().GetField("isSearchResult", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
It'd be better not to use private reflection.  Doing so makes the test much more brittle and less portable.